anesthetic/weighted_pandas.py

```diff
@@ -55,11 +55,18 @@ def __init__(self, *args, **kwargs):
 
     def mean(self):
         """Weighted mean of the sampled distribution."""
-        return np.average(self, weights=self.weight)
+        if self.weight is None:
+            return np.average(self)
+        nonzero = self.weight > 0
+        return np.average(self[nonzero], weights=self.weight[nonzero])
```

Comment:
The failing code coverage comes from the `if.self.weight is None` case.

Do we need to check for this? Or are we confident that any `weight=None` is turned into `weight=1` by the weighted objects?

Reply:
Tests run fine upon deleting those lines...

Reply:
I think that this comes from an earlier version of the code when weights were treated slightly differently. `weight` should now never be `None` (if `_weight==None`, then `weight` is returned as a set of appropriate `1.`s). We should drop it.